sklearn/utils/tests/test_sparsefuncs.py

```diff
@@ -443,6 +443,19 @@ def test_count_nonzero():
     assert_raises(TypeError, count_nonzero, X_csc)
     assert_raises(ValueError, count_nonzero, X_csr, axis=2)
 
+    assert (count_nonzero(X_csr, axis=0).dtype ==
+            count_nonzero(X_csr, axis=1).dtype)
+    assert (count_nonzero(X_csr, axis=0, sample_weight=sample_weight).dtype ==
+            count_nonzero(X_csr, axis=1, sample_weight=sample_weight).dtype)
+
+    # Check dtypes with large sparse matrices too
+    X_csr = sp.csr_matrix((X_csr.data, X_csr.indices.astype(np.int64),
+                           X_csr.indptr.astype(np.int64)), shape=X_csr.shape)
```

Comment:
If you want to test this, you would have to monkeypatch `X_csr.indices` etc as otherwise `csr_matrix` will downcast them as needed (here to `int32`)